include/science/libndtp/utils.h

```diff
@@ -36,7 +36,7 @@ template <typename T>
 std::tuple<ByteArray, BitOffset, bool> to_bytes(
     const std::vector<T>& values,
     uint8_t bit_width,
-    const ByteArray& existing = {},
+    ByteArray& existing = {},
```

Comment:
```suggestion
```

Reply:
using an input reference parameter here was an intentional choice to allow for a ByteArray to be passed in and constructed across several calls to to_bytes(). What is your logic behind removing it?

Reply:
derp, lost context, you're right